### PR TITLE
Feat:(issue_10) Move providers into separate paths

### LINKED ISSUE #10: feature: provider based packaging
Hello,

Looking at this repo while trying CLI V3, I noticed that all altsrc are on the same package.
Since you moved these sources to a separate repository, I assume you would want to add more providers.

Because of this I would expect to not have to import all supported config providers if I don't need them.
I.E. vault, aws secrets, one password, etc.

Would you be interested in this change? 

Keep up the great work.

### PR DESCRIPTION
**Pull request by dearchap**
Fixes #10

### CODE REVIEW

**Comment by YvanDaSilva** [comment]
@dearchap This is awesome!
IMHO, it's going to make programs that depend on urface/cli lighter and better.

**Comment by meatballhat** [comment]
Overall, I am in favor of this direction 🎉 That being said, I haven't worked with nested modules enough to know the trade-offs. I think that given we're in a prerelease phase anyway, I would advocate for each module being in its own repository.

This could also include addressing the awkwardness of namespace collisions so that instead of having to deal with package import aliases, the declared package names could be like `altsrcyaml` `altsrcjson`.

**Comment by dearchap** [comment]
Here's a good example. https://github.com/knadh/koanf

**Comment by dearchap** [comment]
In keeping with koanf example above I have raised a PR in cli

https://github.com/urfave/cli/pull/2009

This allows map value sources to plugin easily without having to do custom lookups. Once that PR is approved the providers here neednt import the root urfave/cli-altsrc and then each provider can plugin to cli directly. How the provider retrieves its information and unmarshals can be left to its implementation

**Comment by mrueg** [comment]
Would you be able to create a new release with these changes in place?

File: json/go.mod [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module github.com/urfave/cli-altsrc/json
```
Comment (mrueg) [inline json/go.mod:1]: @dearchap should these be also on v3 or do you want to keep them loosely coupled?

**Pull request merged**